apps/dotcom/client/src/tla/components/TlaEditor/TlaSnapshotsEditor.tsx

```diff
@@ -0,0 +1,83 @@
+import { getLicenseKey } from '@tldraw/dotcom-shared'
+import { useMemo, useRef } from 'react'
+import { SerializedSchema, TLComponents, TLRecord, Tldraw, usePassThroughWheelEvents } from 'tldraw'
+import { ShareButton } from '../../../components/ShareButton'
+import { ThemeUpdater } from '../../../components/ThemeUpdater/ThemeUpdater'
+import { useLegacyUrlParams } from '../../../hooks/useLegacyUrlParams'
+import { assetUrls } from '../../../utils/assetUrls'
+import { globalEditor } from '../../../utils/globalEditor'
+import { useSharing } from '../../../utils/sharing'
+import { useFileSystem } from '../../../utils/useFileSystem'
+import { useHandleUiEvents } from '../../../utils/useHandleUiEvent'
+import { useMaybeApp } from '../../hooks/useAppState'
+import { TlaPublishFileShareMenu } from '../TlaFileShareMenu/TlaPublishFileShareMenu'
+import { SneakyDarkModeSync } from './SneakyDarkModeSync'
+import { TlaEditorTopLeftPanel } from './TlaEditorTopLeftPanel'
+import styles from './editor.module.css'
+
+const components: TLComponents = {
+	ErrorFallback: ({ error }) => {
+		throw error
+	},
+	SharePanel: () => {
+		const ref = useRef<HTMLDivElement>(null)
+		usePassThroughWheelEvents(ref)
+		return (
+			<div ref={ref} className={styles.topRightPanel}>
+				<TlaPublishFileShareMenu>
+					<ShareButton title={'share-menu.title'} label={'share-menu.title'} />
+				</TlaPublishFileShareMenu>
+			</div>
+		)
+	},
+	MenuPanel: () => {
+		const app = useMaybeApp()
+		return <TlaEditorTopLeftPanel isAnonUser={!app} />
+	},
+}
+
+interface TlaSnapshotEditorProps {
+	schema: SerializedSchema
+	records: TLRecord[]
+}
+
+export function TlaSnapshotsEditor({ schema, records }: TlaSnapshotEditorProps) {
+	// make sure this runs before the editor is instantiated
+	useLegacyUrlParams()
+
+	const handleUiEvent = useHandleUiEvents()
+	const sharingUiOverrides = useSharing()
+	const fileSystemUiOverrides = useFileSystem({ isMultiplayer: true })
+
+	const snapshot = useMemo(
+		() => ({
+			schema,
+			store: Object.fromEntries(records.map((record) => [record.id, record])),
+		}),
+		[schema, records]
+	)
+
+	return (
+		<div className={styles.editor}>
+			<Tldraw
+				licenseKey={getLicenseKey()}
+				assetUrls={assetUrls}
+				snapshot={snapshot}
+				overrides={[sharingUiOverrides, fileSystemUiOverrides]}
+				onUiEvent={handleUiEvent}
+				onMount={(editor) => {
+					;(window as any).app = editor
+					;(window as any).editor = editor
+					editor.updateInstanceState({ isReadonly: true })
```

Comment:
this makes me think we should have a readonly property at the top-level - feels somewhat dirty in general that we have to wait until `onMount` to mark it as readonly... 🤔

Reply:
Yeah, maybe. Think we did have it a while back, but unsure why it was removed.